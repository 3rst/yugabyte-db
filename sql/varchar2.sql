--
-- test type modifier related rules
--

-- ERROR (typmod >= 1)
CREATE TABLE foo (a VARCHAR2(0));

-- ERROR (number of typmods = 1)
CREATE TABLE foo (a VARCHAR2(10, 1));

-- OK
CREATE TABLE foo (a VARCHAR(5000));

-- cleanup
DROP TABLE foo;

-- OK
CREATE TABLE foo (a VARCHAR2(5));

--
-- test that no value longer than maxlen is allowed
--

-- ERROR (length > 5)
INSERT INTO foo VALUES ('abcdef');

-- ERROR (length > 5);
-- VARCHAR2 does not truncate blank spaces on implicit coercion
INSERT INTO foo VALUES ('abcde  ');

-- OK
INSERT INTO foo VALUES ('abcde');

-- OK
INSERT INTO foo VALUES ('abcdef'::VARCHAR2(5));

-- OK
INSERT INTO foo VALUES ('abcde  '::VARCHAR2(5));

--OK
INSERT INTO foo VALUES ('abc'::VARCHAR2(5));

--
-- test whitespace semantics on comparison
--

-- equal
SELECT 'abcde   '::VARCHAR2(10) = 'abcde   '::VARCHAR2(10);

-- not equal
SELECT 'abcde  '::VARCHAR2(10) = 'abcde   '::VARCHAR2(10);


