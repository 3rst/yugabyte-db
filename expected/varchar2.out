--
-- test type modifier related rules
--
-- ERROR (typmod >= 1)
CREATE TABLE foo (a VARCHAR2(0));
ERROR:  length for type varchar must be at least 1
LINE 1: CREATE TABLE foo (a VARCHAR2(0));
                            ^
-- ERROR (number of typmods = 1)
CREATE TABLE foo (a VARCHAR2(10, 1));
ERROR:  invalid type modifier
LINE 1: CREATE TABLE foo (a VARCHAR2(10, 1));
                            ^
-- OK
CREATE TABLE foo (a VARCHAR(5000));
-- cleanup
DROP TABLE foo;
-- OK
CREATE TABLE foo (a VARCHAR2(5));
--
-- test that no value longer than maxlen is allowed
--
-- ERROR (length > 5)
INSERT INTO foo VALUES ('abcdef');
ERROR:  input value length is 6; too long for type varchar2(5)
-- ERROR (length > 5);
-- VARCHAR2 does not truncate blank spaces on implicit coercion
INSERT INTO foo VALUES ('abcde  ');
ERROR:  input value length is 7; too long for type varchar2(5)
-- OK
INSERT INTO foo VALUES ('abcde');
-- OK
INSERT INTO foo VALUES ('abcdef'::VARCHAR2(5));
-- OK
INSERT INTO foo VALUES ('abcde  '::VARCHAR2(5));
--OK
INSERT INTO foo VALUES ('abc'::VARCHAR2(5));
--
-- test whitespace semantics on comparison
--
-- equal
SELECT 'abcde   '::VARCHAR2(10) = 'abcde   '::VARCHAR2(10);
 ?column? 
----------
 t
(1 row)

-- not equal
SELECT 'abcde  '::VARCHAR2(10) = 'abcde   '::VARCHAR2(10);
 ?column? 
----------
 f
(1 row)

